pallets/registrar/src/benchmarks.rs

```diff
@@ -179,5 +179,53 @@ mod benchmarks {
         Pallet::<T>::set_boot_nodes(RawOrigin::Signed(caller), Default::default(), boot_nodes);
     }
 
+    #[benchmark]
+    fn pause_container_chain(y: Linear<1, 50>) {
+        let storage = vec![(vec![1; 4], vec![1; 3_000_000usize]).into()];
+        let storage = new_genesis_data(storage);
+
+        // Deregister all the existing chains to avoid conflicts with the new ones
+        for para_id in Pallet::<T>::registered_para_ids() {
+            Pallet::<T>::deregister(RawOrigin::Root.into(), para_id).unwrap();
+        }
+
+        for i in 0..y {
+            // Twice the deposit just in case
+            let (caller, _deposit_amount) =
+                create_funded_user::<T>("caller", i, T::DepositAmount::get());
+            Pallet::<T>::register(
+                RawOrigin::Signed(caller.clone()).into(),
+                i.into(),
+                storage.clone(),
+            )
+            .unwrap();
+            Pallet::<T>::mark_valid_for_collating(RawOrigin::Root.into(), i.into()).unwrap();
+        }
```

Comment:
I believe the worst case for this benchmark is when both `pending_verification` and `registered_para_ids` are full. So what do you think about adding another for loop like this one but without `mark_valid_for_collating`?
So that paraIds `0..=50` are in `registered_para_ids` and paraIds `1000..=1050` are in `pending_verification` as an example. And then paraId 49 is paused.

And the same applies to the `mark_valid_for_collating` benchmark, it is missing this small extra cost.

Reply:
Yes! Agree on this. Will add them.